### PR TITLE
fix(pie): Process series data-Array . close #11247

### LINKED ISSUE #11247: 饼图数据显示问题
### Version
4.3.0

### Reproduction link
[https://gallery.echartsjs.com/editor.html?c=xbRFTNETY-&v=1](https://gallery.echartsjs.com/editor.html?c=xbRFTNETY-&v=1)

### Steps to reproduce
饼图是支持输入数组数据，但并没有处理数组数据

### What is expected?
希望饼图能将数组的中数据求和，在鼠标悬停时，显示详细的数据

### What is actually happening?
在数据为数组时，饼图只讲其中一个数据进行计算

### PR DESCRIPTION
**Pull request by karma2014**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

To support pie chart series.data to reduce Array .

### Before: What was the problem?

Pie chart series.data can not support to process Array .

### After: How is it fixed in this PR?

When pie chart init , process the series data to support Array data. 

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

NA.

### Merging options

- [x] Please squash the commits into a single one when merge.

### CODE REVIEW

File: src/chart/pie/PieSeries.js [hunk 1]
```diff
@@ -45,6 +48,28 @@ var PieSeries = echarts.extendSeriesModel({
...
+            if(data[i].value instanceof Array){
+                //visualMap component fix me
+                // if(data[i].value[0] instanceof Array){
+                //     if(option.visualMap && option.visualMap instanceof Array){
+                //         for(var j = 0; j < option.visualMap.length; j++){
+                //             option.visualMap[j]['value'] = data[i].value[j][1]; //set second number to visualMap...
+                //         }
+                //     }
+                // }
+                data[i].value = data[i].value.reduce(function(a, b) {
```
Comment (pissang) [inline src/chart/pie/PieSeries.js]: Please do not use `reduce` function. It's not compatitable in IE8

File: src/chart/pie/PieSeries.js [hunk 2]
```diff
@@ -45,6 +48,28 @@ var PieSeries = echarts.extendSeriesModel({
         this._defaultLabelLine(option);
     },
 
+    processOptionSeriesValue: function (option) {
+        var data = option.data;
+
+        for(var i = 0; i < data.length; i++){
+            if(data[i].value instanceof Array){
+                //visualMap component fix me
+                // if(data[i].value[0] instanceof Array){
```
Comment (pissang) [inline src/chart/pie/PieSeries.js]: What's this commented code about?

Comment (karma2014) [reply src/chart/pie/PieSeries.js]: My answer is incorrect.I am trying to fix it.I want sum array value before compute each Sector ‘s area，please wait for my resubmit.

File: src/chart/pie/PieSeries.js (hunk 1 above)
Comment (karma2014) [reply src/chart/pie/PieSeries.js]: Thanks for the reminder， I am try to fix.

**Comment by karma2014** [comment]
My first commit is not correct,so I resubmit to fix #11247 to support pie chart series.data to process Array  data.

File: src/chart/pie/pieLayout.js [hunk 3]
```diff
@@ -68,7 +68,40 @@ export default function (seriesType, ecModel, api, payload) {
...
         });
 
-        var sum = data.getSum(valueDim);
+        var sum = 0;
+        var max = data._rawExtent[1];   
+        var tempMax = 0;                
+        var trueValueArray = [];        
+        var extent = data.getDataExtent(valueDim);
+
+        if(data && data._rawData && data._rawData._data && data._rawData._data.length && data._rawData._data.length > 0){
```
Comment (pissang) [inline src/chart/pie/pieLayout.js]: Property with `_` prefix is private. Please do not access them outside.

**Pull request closed without merging**